src/appleseed/renderer/modeling/scene/basegroup.cpp

```diff
@@ -98,6 +98,16 @@ ShaderGroupContainer& BaseGroup::shader_groups() const
     return impl->m_shader_groups;
 }
 
+void BaseGroup::clear()
+{
+    impl->m_colors.clear();
+    impl->m_textures.clear();
+    impl->m_texture_instances.clear();
+    impl->m_shader_groups.clear();
+    impl->m_assemblies.clear();
```

Comment:
Do we acknowledge here that we're not calling `clear()` on sub-assemblies?

Reply:
We are simply deleting sub-assemblies.